File: MBE.Domain.Elections/PremiumOverride/NonAgeBandedPremiumOverrideCalculator.cs
using MBE.Domain.Elections.DataAccess;
using MBE.Domain.Elections.Models;

namespace MBE.Domain.Elections.PremiumOverride
{
    public interface INonAgeBandedPremiumOverrideCalculator
    {
        decimal GetPremiumOverride(ElectionData electionData);
    }

    public class NonAgeBandedPremiumOverrideCalculator : INonAgeBandedPremiumOverrideCalculator
    {
        private readonly ITierCostRepository m_tierCostRepository;
        public NonAgeBandedPremiumOverrideCalculator(ITierCostRepository tierCostRepository)
        {
            m_tierCostRepository = tierCostRepository;
        }

        public decimal GetPremiumOverride(ElectionData electionData)
        {
            var tier = m_tierCostRepository.SelectTierCost(electionData.TierID);
            return tier.PremiumOverride;
        }

    }
}

File: MBE.Domain.Elections.Tests/NonAgeBandedPremiumOverrideCalculator.fs
namespace MBE.Domain.Elections.Tests
open MBE.Domain.Elections
open MBE.Domain.Elections.Models
open System
open Xunit
open FsUnit.Xunit
open MBE.Domain.Elections.DataAccess
open MBE.Domain.Elections.PremiumOverride

module NonAgeBandedPremiumOverrideCalculator =

    [<Fact>]
    let ``Premium Override from TierCost is returned Test``() =
        let tier = new TierCost(TierID = 1, BaseTierID = 1, PremiumOverride = decimal 100.00, FlatRate = true )
        let tierCostRepository = { new ITierCostRepository with member this.SelectTierCost(x) = tier }
        let electionData = new ElectionData (ParentUserID = 1500, PlanID = 1, TierID = 1)

        let actual = (new NonAgeBandedPremiumOverrideCalculator(tierCostRepository)).GetPremiumOverride(electionData)

        Assert.Equal(decimal 100.00, actual)